src/core/thread/thread_netif.cpp

```diff
@@ -171,6 +172,28 @@ ThreadError ThreadNetif::RouteLookup(const Ip6::Address &source, const Ip6::Addr
     return error;
 }
 
+ThreadError ThreadNetif::TmfFilter(const otMessage *aMessage, const otMessageInfo *aMessageInfo)
```

Comment:
If we want to make this a more general solution, then I think we should move this kind of "fire walling" behavior down to the IPv6 layer.  I'm fine with the current implementation, but would not want to project this through the CoAP API.